rosetta-ide/src/main/java/com/regnosys/rosetta/ide/inlayhints/IInlayHintsResolver.java

```diff
@@ -1,3 +1,19 @@
+/*
+ * Copyright 2024 REGnosys
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
```

Comment:
Another note: should this banner also be added to Xtend files? (right now I only see them on Java files)

What about our `.xtext`, `.xsemantics` and `.xcore` files?

Reply:
The more the code logic is important, the more is susceptible to be used, the more the banner is important.

Configuration files may not be covered, assuming they do not have valuable Intellectual Property in it, however some teams prefer to add the banner across ll files.

Last note, some files, like JSON, do not allow comments, and therefore the banner cannot be added there; feel free to take notes on how this works into the `NOTICE` file